Marlin/src/libs/MAX31865.cpp

```diff
@@ -270,82 +265,84 @@ void MAX31865::initFixedFlags(max31865_numwires_t wires) {
  */
 uint16_t MAX31865::readRaw() {
 
-  uint16_t rtd;
-
-#if DISABLED(MAX31865_USE_AUTO_MODE)
-
-  if (PENDING(millis(), nextEventStamp)) {
-    DEBUG_ECHOLNPGM("MAX31865 waiting for event ", nextEvent);
-    return lastRead;
-  }
-
-  switch (nextEvent) {
-    case SETUP_BIAS_VOLTAGE:
-
-      enableBias();
+  const millis_t now = millis();
 
-      nextEventStamp = millis() + 11; // wait at least 11msec before enabling 1shot
-      nextEvent = SETUP_1_SHOT_MODE;
+  #if ENABLED(MAX31865_USE_AUTO_MODE)
 
-      DEBUG_ECHOLN("MAX31865 bias voltage enabled");
+    const uint16_t rtd = readRegister16(MAX31865_RTDMSB_REG);
+    DEBUG_ECHOLNPGM("MAX31865 RTD MSB:", (rtd >> 8), " LSB:", (rtd & 0x00FF));
 
-      break;
-
-    case SETUP_1_SHOT_MODE:
-
-      oneShot();
-
-      nextEventStamp = millis() + 65; // wait at least 65msec before reading RTD register
-      nextEvent = READ_RTD_REG;
-
-      DEBUG_ECHOLN("MAX31865 1 shot mode enabled");
-
-      break;
-
-    case READ_RTD_REG:
-
-#endif // DISABLED(MAX31865_USE_AUTO_MODE)
-
-      rtd = readRegister16(MAX31865_RTDMSB_REG);
-
-      DEBUG_ECHOLNPGM("MAX31865 RTD MSB:", (rtd >> 8), " LSB:", (rtd & 0x00FF));
-
-    #if ENABLED(MAX31865_USE_READ_ERROR_DETECTION) || DISABLED(MAX31865_USE_AUTO_MODE)
-      millis_t now = millis();
-    #endif
-
-      if (rtd & 1) {
-        lastFault = readRegister8(MAX31865_FAULTSTAT_REG);
-        lastRead |= 1;
-        clearFault(); // also clears the bias voltage flag, so no further action is required
-
-        DEBUG_ECHOLNPGM("MAX31865 read fault: ", rtd);
-      }
+    if (rtd & 1) {
+      lastFault = readRegister8(MAX31865_FAULTSTAT_REG);
+      lastRead |= 1;
+      clearFault(); // also clears the bias voltage flag, so no further action is required
+      DEBUG_ECHOLNPGM("MAX31865 read fault: ", rtd);
+    }
     #if ENABLED(MAX31865_USE_READ_ERROR_DETECTION)
       else if (ABS(lastRead - rtd) > 500 && PENDING(now, lastReadStamp + 1000)) { // if two readings within a second differ too much (~20°C), consider it a read error.
         lastFault = 0x01;
         lastRead |= 1;
-
         DEBUG_ECHOLNPGM("MAX31865 read error: ", rtd);
       }
     #endif
-      else {
-        lastRead = rtd;
-      #if ENABLED(MAX31865_USE_READ_ERROR_DETECTION)
-        lastReadStamp = now;
-      #endif
-      }
-
-#if DISABLED(MAX31865_USE_AUTO_MODE)
-      if (!(rtd & 1)) { // if clearFault() was not invoked, need to clear the bias voltage and 1-shot flags
-        resetFlags();
-      }
+    else {
+      lastRead = rtd;
+      TERN_(MAX31865_USE_READ_ERROR_DETECTION, lastReadStamp = now);
+    }
+
+  #else
+
+    if (PENDING(now, nextEventStamp)) {
+      DEBUG_ECHOLNPGM("MAX31865 waiting for event ", nextEvent);
+      return lastRead;
+    }
+
+    switch (nextEvent) {
+      case SETUP_BIAS_VOLTAGE:
+        enableBias();
+        nextEventStamp = now + 11; // wait at least 11msec before enabling 1shot
```

Comment:
I have an objection here. Time counting for the next event should start *after* the flag is set, not before. I know it's a minor thing, the time difference is negligible, probably will never matter in practice, but conceptually this doesn't feel right, a new timestamp should be used here (same applies for line 310 below).

Reply:
The question is whether any code executed since assigning `now` takes more than a couple milliseconds. If it does then we should take the latest value of `millis()` here, otherwise `now` should be sufficient.

Reply:
I'm pretty sure it doesn't take more than a few μsec, although changing the flags involves SPI operations, which do include some variable delays. But it's still a matter of semantics, this way it's clear in the code that the time is measured from the point after setting the flags. I think that alone is worth the cost of a second invocation of `millis()`.

Reply:
Just for the kicks, I measured it on my LPC1768 board, and it takes 17μsec to run lines 292-299. Still, maintaining code clarity is more important than the occasional 0.15μsec spent on a `millis()` call.